src/api/v1beta1/properties.go

```diff
@@ -387,3 +406,30 @@ func splitArg(arg string) (key, value string) {
 	value = split[1]
 	return
 }
+
+func getRawImageTag(imageURI string) string {
+	if !strings.Contains(imageURI, ":") {
+		return "latest"
+	}
+	splitURI := strings.Split(imageURI, ":")
+	return splitURI[len(splitURI)-1]
+}
+
+// getSafeImageTag extracts and normalizes the provided image's tag
+func getSafeImageTag(imageURI string) string {
+	rawTag := getRawImageTag(imageURI)
+	return normalizeImageTag(rawTag)
+}
+
+// normalizeImageTag normalizes the tag
+// so it matches the requirements described here: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+func normalizeImageTag(tag string) string {
+	notAllowedCharsRegex := regexp.MustCompile(`[^\d\w-_.]`)
+	safeTag := notAllowedCharsRegex.ReplaceAllString(tag, "-")
+	if len(safeTag) > maxLabelValueLength {
+		safeTag = safeTag[:maxLabelValueLength]
+	}
+	notAllowedEndingRegex := regexp.MustCompile(`[^\w\d]+$`)
+	safeTag = notAllowedEndingRegex.ReplaceAllString(safeTag, "")
+	return safeTag
+}
```

Comment:
as mentioned before as a comment, those functions should be in a separate package